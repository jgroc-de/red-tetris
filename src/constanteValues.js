export const numberOfLines = 20
export const numberOfColumns = 10
export const filledSquared = '-'
export const filledTetriminosSquared = 'x'
export const emptySquared = ' '
export const rotationCases = 4
export const LShape = [
    [
        emptySquared + emptySquared + filledTetriminosSquared,
        filledTetriminosSquared + filledTetriminosSquared + filledTetriminosSquared,
        emptySquared + emptySquared + emptySquared,
    ],
    [
        emptySquared + filledTetriminosSquared + emptySquared,
        emptySquared + filledTetriminosSquared + emptySquared,
        emptySquared + filledTetriminosSquared + filledTetriminosSquared,
    ],
    [
        emptySquared + emptySquared + emptySquared,
        filledTetriminosSquared + filledTetriminosSquared + filledTetriminosSquared,
        filledTetriminosSquared + emptySquared + emptySquared,
    ],
    [
        filledTetriminosSquared + filledTetriminosSquared + emptySquared,
        emptySquared + filledTetriminosSquared + emptySquared,
        emptySquared + filledTetriminosSquared + emptySquared,
    ]
]
export const JShape = [
    [
        filledTetriminosSquared + emptySquared + emptySquared,
        filledTetriminosSquared + filledTetriminosSquared + filledTetriminosSquared,
        emptySquared + emptySquared + emptySquared,
    ],
    [
        emptySquared + filledTetriminosSquared + filledTetriminosSquared,
        emptySquared + filledTetriminosSquared + emptySquared,
        emptySquared + filledTetriminosSquared + emptySquared,
    ],
    [
        emptySquared + emptySquared + emptySquared,
        filledTetriminosSquared + filledTetriminosSquared + filledTetriminosSquared,
        emptySquared + emptySquared + filledTetriminosSquared,
    ],
    [
        emptySquared + filledTetriminosSquared + emptySquared,
        emptySquared + filledTetriminosSquared + emptySquared,
        filledTetriminosSquared + filledTetriminosSquared + emptySquared,
    ]
]
export const SShape = [
    [
        emptySquared + filledTetriminosSquared + filledTetriminosSquared,
        filledTetriminosSquared + filledTetriminosSquared + emptySquared,
        emptySquared + emptySquared + emptySquared,
    ],
    [
        emptySquared + filledTetriminosSquared + emptySquared,
        emptySquared + filledTetriminosSquared + filledTetriminosSquared,
        emptySquared + emptySquared + filledTetriminosSquared,
    ],
    [
        emptySquared + emptySquared + emptySquared,
        emptySquared + filledTetriminosSquared + filledTetriminosSquared,
        filledTetriminosSquared + filledTetriminosSquared + emptySquared,
    ],
    [
        filledTetriminosSquared + emptySquared + emptySquared,
        filledTetriminosSquared + filledTetriminosSquared + emptySquared,
        emptySquared + filledTetriminosSquared + emptySquared,
    ]
]
export const OShape = [
    [
        emptySquared + filledTetriminosSquared + filledTetriminosSquared + emptySquared,
        emptySquared + filledTetriminosSquared + filledTetriminosSquared + emptySquared,
        emptySquared + emptySquared + emptySquared + emptySquared,
        emptySquared + emptySquared + emptySquared + emptySquared,
    ],
    [
        emptySquared + filledTetriminosSquared + filledTetriminosSquared + emptySquared,
        emptySquared + filledTetriminosSquared + filledTetriminosSquared + emptySquared,
        emptySquared + emptySquared + emptySquared + emptySquared,
        emptySquared + emptySquared + emptySquared + emptySquared,
    ],
    [
        emptySquared + filledTetriminosSquared + filledTetriminosSquared + emptySquared,
        emptySquared + filledTetriminosSquared + filledTetriminosSquared + emptySquared,
        emptySquared + emptySquared + emptySquared + emptySquared,
        emptySquared + emptySquared + emptySquared + emptySquared,
    ],
    [
        emptySquared + filledTetriminosSquared + filledTetriminosSquared + emptySquared,
        emptySquared + filledTetriminosSquared + filledTetriminosSquared + emptySquared,
        emptySquared + emptySquared + emptySquared + emptySquared,
        emptySquared + emptySquared + emptySquared + emptySquared,
    ],
]
export const TShape = [
    [
        emptySquared + filledTetriminosSquared + emptySquared,
        filledTetriminosSquared + filledTetriminosSquared + filledTetriminosSquared,
        emptySquared + emptySquared + emptySquared,
    ],
    [
        emptySquared + filledTetriminosSquared + emptySquared,
        emptySquared + filledTetriminosSquared + filledTetriminosSquared,
        emptySquared + filledTetriminosSquared + emptySquared,
    ],
    [
        emptySquared + emptySquared + emptySquared,
        filledTetriminosSquared + filledTetriminosSquared + filledTetriminosSquared,
        emptySquared + filledTetriminosSquared + emptySquared,
    ],
    [
        emptySquared + filledTetriminosSquared + emptySquared,
        filledTetriminosSquared + filledTetriminosSquared + emptySquared,
        emptySquared + filledTetriminosSquared + emptySquared,
    ]
]
export const ZShape = [
    [
        filledTetriminosSquared + filledTetriminosSquared + emptySquared,
        emptySquared + filledTetriminosSquared + filledTetriminosSquared,
        emptySquared + emptySquared + emptySquared,
    ],
    [
        emptySquared + emptySquared + filledTetriminosSquared,
        emptySquared + filledTetriminosSquared + filledTetriminosSquared,
        emptySquared + filledTetriminosSquared + emptySquared,
    ],
    [
        emptySquared + emptySquared + emptySquared,
        filledTetriminosSquared + filledTetriminosSquared + emptySquared,
        emptySquared + filledTetriminosSquared + filledTetriminosSquared,
    ],
    [
        emptySquared + filledTetriminosSquared + emptySquared,
        filledTetriminosSquared + filledTetriminosSquared + emptySquared,
        filledTetriminosSquared + emptySquared + emptySquared,
    ],

]
export const IShape = [
    [
        emptySquared + emptySquared + emptySquared + emptySquared,
        filledTetriminosSquared + filledTetriminosSquared + filledTetriminosSquared + filledTetriminosSquared,
        emptySquared + emptySquared + emptySquared + emptySquared,
        emptySquared + emptySquared + emptySquared + emptySquared,
    ],
    [
        emptySquared + emptySquared + filledTetriminosSquared + emptySquared,
        emptySquared + emptySquared + filledTetriminosSquared + emptySquared,
        emptySquared + emptySquared + filledTetriminosSquared + emptySquared,
        emptySquared + emptySquared + filledTetriminosSquared + emptySquared,
    ],
    [
        emptySquared + emptySquared + emptySquared + emptySquared,
        emptySquared + emptySquared + emptySquared + emptySquared,
        filledTetriminosSquared + filledTetriminosSquared + filledTetriminosSquared + filledTetriminosSquared,
        emptySquared + emptySquared + emptySquared + emptySquared,
    ],
    [
        emptySquared + filledTetriminosSquared + emptySquared + emptySquared,
        emptySquared + filledTetriminosSquared + emptySquared + emptySquared,
        emptySquared + filledTetriminosSquared + emptySquared + emptySquared,
        emptySquared + filledTetriminosSquared + emptySquared + emptySquared,
    ],
]
