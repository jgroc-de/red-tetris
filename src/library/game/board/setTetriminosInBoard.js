import insertInBoard from './insertInBoard'

function setTetriminosInBoard(fullBoard, shape, position, numberOfColumns, emptySquared) {
    let updatedBoard = fullBoard

    for (let y = 0; y < shape.length; y++) {
        for (let x = 0; x < shape[0].length; x++) {
            let posY = (position.y + y) * numberOfColumns
            let posX = position.x + x
            if (shape[y][x] !== emptySquared) {
                updatedBoard = insertInBoard(updatedBoard, posX + posY, shape[y][x])
            }
        }
    }
    return updatedBoard
}

export default setTetriminosInBoard
