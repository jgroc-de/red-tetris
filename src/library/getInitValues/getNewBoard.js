function getNewBoard(length, emptySquared) {
    return emptySquared.repeat(length)
}

export default getNewBoard
