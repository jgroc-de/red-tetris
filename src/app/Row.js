import { numberOfColumns } from '../constanteValues'
import Square from './Square'

function Row(props) {
    const squares = []
    for (let i = 0; i < numberOfColumns; i++) {
        squares.push(<Square key={i} square={props.squares[i]}/>)
    }

    return (
        <div key={props.index} className='is-primary'>
            {squares}
        </div>
    )
}

export default Row
