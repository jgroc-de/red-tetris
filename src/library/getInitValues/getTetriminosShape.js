import { IShape, JShape, LShape, OShape, SShape, TShape, ZShape } from "../../constanteValues"

function getTetriminosShape(shape) {
    switch (shape) {
        case 'I':
            return IShape
        case 'J':
            return JShape
        case 'L':
            return LShape
        case 'O':
            return OShape
        case 'S':
            return SShape
        case 'T':
            return TShape
        case 'Z':
            return ZShape
        default:
            return OShape
    }
}

export default getTetriminosShape