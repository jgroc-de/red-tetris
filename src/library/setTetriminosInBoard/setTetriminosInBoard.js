import {numberOfColumns, emptySquared} from '../../constanteValues'
import insertInBoard from './insertInBoard'

function setTetriminosInBoard(fullBoard, shape, position, token) {
    let updatedBoard = fullBoard

    for (let y = 0; y < shape.length; y++) {
        for (let x = 0; x < shape[0].length; x++) {
            let posY = (position.y + y) * numberOfColumns
            let posX = position.x + x
            if (shape[y][x] !== emptySquared) {
                updatedBoard = insertInBoard(updatedBoard, posX + posY, token)
            }
        }
    }
    return updatedBoard
}

export default setTetriminosInBoard
