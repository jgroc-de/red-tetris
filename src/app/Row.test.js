import React from 'react'
import renderer from 'react-test-renderer'
import Row from './Row'
import  {numberOfColumns, numberOfLines, filledTetriminosSquared, emptySquared} from '../constanteValues'
import setTetriminosInBoard from '../library/setTetriminosInBoard/setTetriminosInBoard'
import getNewBoard from '../library/getInitValues/getNewBoard'
import getTetriminosShape from '../library/getInitValues/getTetriminosShape'

test('empty row', () => {
    let board = getNewBoard(numberOfLines * numberOfColumns, emptySquared)
    let index = 10
    const component = renderer.create(<Row
        key={index}
        number={index}
        squares={board.slice(index * numberOfColumns, index * numberOfColumns + numberOfColumns)}
    />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})

test('row containing tetriminos', () => {
    let board = getNewBoard(numberOfLines * numberOfColumns, emptySquared)
    let index = 4
    let tetriminosShape = getTetriminosShape('O')[0]
    let tetriminosPosition = { x: 4, y: 3 }
    board = setTetriminosInBoard(board, tetriminosShape, tetriminosPosition, filledTetriminosSquared)
    const component = renderer.create(<Row
        key={index}
        number={index}
        squares={board.slice(index * numberOfColumns, index * numberOfColumns + numberOfColumns)}
    />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})
