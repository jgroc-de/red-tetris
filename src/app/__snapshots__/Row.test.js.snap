// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`empty row 1`] = `
<div
  className="is-primary"
>
  <div
    className="square"
  />
  <div
    className="square"
  />
  <div
    className="square"
  />
  <div
    className="square"
  />
  <div
    className="square"
  />
  <div
    className="square"
  />
  <div
    className="square"
  />
  <div
    className="square"
  />
  <div
    className="square"
  />
  <div
    className="square"
  />
</div>
`;

exports[`row containing tetriminos 1`] = `
<div
  className="is-primary"
>
  <div
    className="square"
  />
  <div
    className="square"
  />
  <div
    className="square"
  />
  <div
    className="square"
  />
  <div
    className="square"
  />
  <div
    className="square colored"
  />
  <div
    className="square colored"
  />
  <div
    className="square"
  />
  <div
    className="square"
  />
  <div
    className="square"
  />
</div>
`;
